ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -490,16 +490,25 @@ function admin_ack() {
         echo "Admin ack is not required in either z-stream upgrade or 4.7 and earlier" && return
     fi
 
-    local out; out="$(oc -n openshift-config-managed get configmap admin-gates -o jsonpath='{.data}')"
+    local out; out="$(oc -n openshift-config-managed get configmap admin-gates -o json | jq -r ".data")"
     if [[ ${out} != *"ack-4.${SOURCE_MINOR_VERSION}"* ]]; then
-        echo "Admin ack not required" && return
+        echo "Admin ack not required: ${out}" && return
     fi
 
-    echo "Require admin ack"
+    echo "Require admin ack:\n ${out}"
     local wait_time_loop_var=0 ack_data
-    ack_data="$(echo ${out} | awk '{print $2}' | cut -f2 -d\")" && echo "Admin ack patch data is: ${ack_data}"
-    oc -n openshift-config patch configmap admin-acks --patch '{"data":{"'"${ack_data}"'": "true"}}' --type=merge
 
+    ack_data="$(echo ${out} | jq -r "keys[]")"
+    for ack in ${ack_data};
+    do
+        # e.g.: ack-4.12-kube-1.26-api-removals-in-4.13
+        if [[ "${ack}" == *4\.$TARGET_MINOR_VERSION ]]
```

Comment:
Here should be `SOURCE_MINOR_VERSION` instead of `TARGET_MINOR_VERSION`, right?

Reply:
target might be more correct? for example in a chain upgrade, 4.13 is the target version while 4.12 is the source version in both of keys.
```
ack-4.11-kube-1.25-api-removals-in-4.12
ack-4.12-kube-1.26-api-removals-in-4.13
```

Reply:
actually here I think it should be `[[ "${ack}" == *"ack-4.${TARGET_MINOR_VERSION}"* ]]` should be more accurate.

Reply:
/s/ack-4.${TARGET_MINOR_VERSION}/ack-4.${SOURCE_MINOR_VERSION}? Agree to ack-***